### PR TITLE
[Storage] add Asjob to Storage Longrunning cmdlets

### PR DESCRIPTION
**Pull request by blueww**
## Description

This PR is to fix https://github.com/Azure/azure-powershell/issues/5409 
This PR still missing changelog and test.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Storage/Storage/Blob/Cmdlet/GetAzureStorageBlobContent.cs [hunk 1]
```diff
@@ -253,7 +253,12 @@ internal void GetBlobContent(CloudBlob blob, string fileName, bool isValidBlob =
         /// <returns>full file path if file path is valid, otherwise throw an exception</returns>
         internal string GetFullReceiveFilePath(string fileName, string blobName, DateTimeOffset? snapshotTime)
         {
-            String filePath = Path.Combine(CurrentPath(), fileName);
```
[the remark this answers is not in the document]
Comment (blueww) [reply src/Storage/Storage/Blob/Cmdlet/GetAzureStorageBlobContent.cs]: The reason for CurrentPath() not work, is: AzurePSCmdlet.BeginProcessing will be called in processRecord, and the this.SessionState will be null.

File: src/Storage/Storage/File/Cmdlet/SetAzureStorageFileContent.cs [hunk 2]
```diff
@@ -73,8 +73,23 @@ public class SetAzureStorageFileContent : StorageFileDataManagementCmdletBase
...
-            FileInfo localFile = new FileInfo(this.GetUnresolvedProviderPathFromPSPath(this.Source));
+            // With -asjob, only absolute path works, so Source must be absolute path.
+            if (AsJob.IsPresent && !System.IO.Path.IsPathRooted(this.Source))
+            {
+                throw new System.ArgumentException(string.Format(Resources.InvalidPathForAsJob, "Source", this.Source), "Source");
+            }
+            string filePath = this.Source;
+            if (!AsJob.IsPresent)
+            {
+                filePath = this.GetUnresolvedProviderPathFromPSPath(this.Source);
```
[the remark this answers is not in the document]
Comment (blueww) [reply src/Storage/Storage/File/Cmdlet/SetAzureStorageFileContent.cs]: Actually GetUnresolvedProviderPathFromPSPath() also not work when use Asjob.

[the remark this answers is not in the document]
Comment (markcowl) [reply src/Storage/Storage/File/Cmdlet/SetAzureStorageFileContent.cs]: If you can show the updated code, I'll take a look.

[the remark this answers is not in the document]
Comment (blueww) [reply src/Storage/Storage/File/Cmdlet/SetAzureStorageFileContent.cs]: @markcowl 
The code is already in this PR with commit https://github.com/Azure/azure-powershell/pull/8741/commits/fdf3f873ecb8fdffe350623e84e1c9879b9108fa

**Comment by MiYanni** [comment]
@markcowl Needs re-review after Wei Wei addressed/replied to comments.

**Comment by blueww** [comment]
@markcowl 
Would you please help to look at the issue in this PR?
Since it's near build, and I still have many other features in hand, how about following 2 options:
1. support -asjob only with absolute path now, and merge the PR. We can add a issue to support the relative path later.
2. If you have time to look the code, could you please find a way to make the relative path resolve work, and the cmdlets runs correct?

**Comment by markcowl** [comment]
@blueww Closing in favor of #8905 Please take a look.

**Pull request closed without merging**